shell/src/main/java/org/apache/accumulo/shell/commands/ScanCommand.java

```diff
@@ -419,6 +438,7 @@ public Options getOptions() {
     outputFileOpt.setArgName("file");
     contextOpt.setArgName("context");
     executionHintsOpt.setArgName("<key>=<value>{,<key>=<value>}");
+    scanServerOpt.setArgName("immediate|eventual");
```

Comment:
In #2411 I called it an Isolated Scanner, but we already have something along those lines. In that issue @keith-turner made a comment which was the genesis for the current name. I'm not tied to the name, do you have a different suggestion on what it should be?

Reply:
The names should imply "use an eventually consistent scan server" instead of "create an eventually consistent scanner". In other words, the name should explicitly declare the resource that the scanner is allowed to use, rather than a characteristic trait of the scanner itself. The scanner characteristics are not inherent to the scanner, but the result of using the particular server type.

For the shell, `--use-scan-server` (with description that says scan servers offer eventual consistency) would be better than `--consistency eventual` on the scanner.

For the Scanner API, perhaps `permitScanServer()`, which allows the scanner to use scan server's, if they are available, with a javadoc that explains that scan servers are eventually consistent.

This changes the way we think about the Scanner object, based on the resources it is using (and characteristics that result from using those resources; eventual consistency is only one such characteristic... other characteristics are also applicable... like potential for less resource contention on the server-side, potentially dynamically scalable, etc.), rather than altering the inherent characteristics of the Scanner object itself.

Reply:
My thinking on the naming is that there are multiple way to implement eventual consistency, with scan servers being one.  From the the API perspective I was thinking a user would declare they are ok with eventual consistency and that runtime configuration would determine how that actually executes.

Reply:
> My thinking on the naming is that there are multiple way to implement eventual consistency, with scan servers being one. From the the API perspective I was thinking a user would declare they are ok with eventual consistency and that runtime configuration would determine how that actually executes.

That makes sense if the desired primary outcome was "eventual consistency". It's not, though. It's merely an acceptable side-effect of what the user is really asking for, and not its essential/sufficient characteristic.

This would be like calling the BatchScanner an UnorderedScanner. There are lots of ways to implement a scanner that returns data out of order. However, the primary characteristic in that case is that it scans batches of ranges in parallel, not that it's unordered. We've previously discussed that it really should be called ParallelScanner instead, because that is its essential characteristic. It is *necessarily* unordered, but it is *sufficient* to describe it as parallel, because all its secondary characteristics, including the fact that it returns results out of order, are merely side-effects of that essential characteristic. The same principle is applicable to the naming for this feature. We should really be naming it based on what is sufficient to describe its essential characteristics, not describing a subset of its necessary characteristics.

Reply:
>That makes sense if the desired primary outcome was "eventual consistency". It's not, though. It's merely an acceptable side-effect of what the user is really asking for, and not its essential/sufficient characteristic.

I agree that a developer will enable eventual consistency because they want better performance and/or availability and they are ok with stale data.  One important thing to consider is that for evaluating code for correctness (like I am reviewing Accumulo code that someone else wrote) is that the consistency level of the data returned by the scanner is extremely important (probably more important than the performance benefits desired by the author).  The fact that a scanner may return stale data needs to be very clearly communicated when considering other names.